src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlExceptionTest.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.Tests
+{
+    public class SqlExceptionTest
+    {
+        private const string badServer = "92B96911A0BD43E8ADA4451031F7E7CF";
+
+        [Fact]
+        [ActiveIssue("12161", TestPlatforms.AnyUnix)]
+        public void SerializationTest()
+        {
+            SqlException e = CreateException();
+            string json = Newtonsoft.Json.JsonConvert.SerializeObject(e);
+
+            var settings = new JsonSerializerSettings()
+            {
+                TypeNameHandling = TypeNameHandling.All,
+            };
+
+            // TODO: Deserialization fails on Unix with "Member 'ClassName' was not found."
+            var sqlEx = Newtonsoft.Json.JsonConvert.DeserializeObject<Microsoft.Data.SqlClient.SqlException>(json, settings);
+            
+            Assert.Equal(e.ClientConnectionId, sqlEx.ClientConnectionId);
+            Assert.Equal(e.StackTrace, sqlEx.StackTrace);
```

Comment:
I think it does not work properly. We have missed `,"NativeErrorCode":67` and `ComponentModel.Win32` in **NetCore**. You can check it by serializing the `sqlEx` again and comparing its result with `json`.
And in **NetFx** we have more differences.

Reply:
I saw these missing elements, but we can't compare as they're not serialized.
Also because `NativeErrorCode` is not accessible, we cannot serialize it either.